onnxruntime/core/providers/cpu/math/clip.h

```diff
@@ -32,4 +32,36 @@ class Clip final : public OpKernel {
   T min_;
 };
 
+template <typename T>
+class Clip final : public OpKernel {
+ public:
+  Clip(const OpKernelInfo& info) : OpKernel(info) {
+  }
+
+  Status Compute(OpKernelContext* ctx) const override {
+    const auto* X = ctx->Input<Tensor>(0);
+    const auto* min = ctx->Input<Tensor>(1);
+    const auto* max = ctx->Input<Tensor>(2);
+    Tensor* Y = ctx->Output(0, X->Shape());
+
+    auto min_val = -std::numeric_limits<T>::infinity();
+    auto max_val = std::numeric_limits<T>::infinity();
+    if (min) {
+      ORT_ENFORCE(min->Shape().NumDimensions() == 0, "min should be a scalar.");
+      min_val = *(min->template Data<T>());
```

Comment:
ONNX generally allows both rank 0, and rank 1 with shape {1} as scalars. Could/should this be min->Shape().Size() == 1?

Reply:
I did have a few discussion with Rama concerning a few other similar ops... The conclusion was to follow the ONNX spec as closely as possible...  In ORT we did allow both in the past and now we cant simply change this check because it will hurt backward compatibility however for new ops we should do strict checks... 

@gramalingam : Comments?